### PR TITLE
Add recursive option for ls command

### PR DESCRIPTION
**Pull request by davecap**

### CODE REVIEW

File: solvebio/cli/data.py [hunk 1]
```diff
@@ -743,6 +743,8 @@ def _ls(full_path):
                 file_.last_modified, file_.object_type.ljust(8), file_.full_path
             )
         )
+        if recursive and file_.object_type == "folder":
+            _ls(file_.full_path + "/*", recursive=True)
 
     if len(files) == 0:
```
Comment (nishanthmerwin) [inline solvebio/cli/data.py:749]: This logic needs to be modified if this is now a recursive function. Currently, if there's any empty directory in the descendents, it prints out the "No files found..." message.

Comment (davecap) [reply solvebio/cli/data.py:749]: Ah, makes sense, thanks

Comment (davecap) [reply solvebio/cli/data.py:749]: Pushed a change to handle this

**Review by nishanthmerwin — APPROVED** [review]
LGTM!

**Comment by davecap** [comment]
@nishanthmerwin can you give the latest a try when you have a few mins

**Comment by nishanthmerwin** [comment]
Tried it on a big folder:

```
❯ solvebio ls "admin-qb-int-dev:Nishanth's Test Vault:/source" --recursive | wc -l
1468
❯ time !!
❯ time solvebio ls "admin-qb-int-dev:Nishanth's Test Vault:/source" --recursive | wc -l
1468
solvebio ls "admin-qb-int-dev:Nishanth's Test Vault:/source" --recursive  0.34s user 0.01s system 2% cpu 16.094 total
wc -l  0.00s user 0.00s system 0% cpu 16.094 total
```

It's not the fastest, but it does the job. Looked through the output and didn't see anything too off.

**Pull request merged**